src/planet-a/icosahedron.h

```diff
@@ -30,12 +30,14 @@
 
 #include <vector>
 
+#include <cstddef>
+
 namespace planeta
 {
 
 // Fundamental geometric properties of an icosahedron
-inline constexpr int const gc_icoVrtxCount = 12;
-inline constexpr int const gc_icoTriCount = 20;
+inline constexpr std::size_t const gc_icoVrtxCount = 12;
+inline constexpr std::size_t const gc_icoTriCount = 20;
```

Comment:
Isn't std::size_t intended data sizes and array indices? This feels very out of place here describing a mathematical property.

Reply:
You're describing a "count" of something. Count's can't be negative and frequently are used as indicies into other objects, or are compared against other objects.

Reply:
How about just an unsigned int? std::size_t conveys an intention that this variable has something important to do with hardware, at least for me.

Reply:
std::size_t has nothing to do with hardware, beyond having the same maximum value as the maximum number of addressable locations in memory on a particular platform.

std::size_t is "the number of things" variable. This is a count of vertices, std::size_t is the right variable type.

Reply:
> std::size_t has nothing to do with hardware, beyond [reason why std::size_t has something to do with hardware]

???

'Counts' can include things like money, score, or number of items acquired, which may not correlate with something in memory. In this case, I don't believe size_t is intended for every count.

Consider other mathematical constants like Avogadro's number (~6.02e23), which is a count of particles in an SI mole. This is a bit of an extreme case but I hope it gets the point across.

I'm not concerned of any performance differences or potential widening/narrowing; simply the intention of what this variable is for. It can be used with array, but it's a general purpose mathematical constant that has nothing to do with hardware/platform/memory.

Reply:
uint8_t better?

Reply:
If I remember correctly, the compiler/analysers should complain less when compared with unsigned values, vs being a signed int before. That should be fine